hotshot-state-prover/src/service.rs

```diff
@@ -423,6 +453,148 @@ pub async fn sync_state<ApiVer: StaticVersionType>(
     let proof_gen_elapsed = Instant::now().signed_duration_since(proof_gen_start);
     tracing::info!("Proof generation completed. Elapsed: {proof_gen_elapsed:.3}");
 
+    Ok((proof, public_input))
+}
+
+async fn epoch_catchup(
+    state: &mut ProverServiceState,
+    provider: &impl Provider,
+    light_client_address: Address,
+    mut cur_st_state: StakeTableState,
```

Comment:
this `mut cur_st_state` looks weird as the func signature.
we should create a local mutable variable instead.